### PR TITLE
 NEW Add GitHub commands to sync labels to supported modules and to check rate limits

### LINKED ISSUE #112: Add command to sync labels to all supported modules
As a core committer
I want to synchronise GitHub labels on all of the supported modules
So that they're always keep up to date

See silverstripe/silverstripe-framework#8343

I've made a start on this with https://github.com/creative-commoners/maintain but figure it may as well be part of cow

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves #112 

This is a port of https://github.com/creative-commoners/maintain which uses centralised modules and labels lists and the GitHub API to push labels out to all supported modules:

* https://github.com/silverstripe/supported-modules/blob/gh-pages/modules.json
* https://github.com/silverstripe/supported-modules/blob/gh-pages/labels.json

It also includes a GitHub rate limit API check command since the sync can exceed your hourly rate limiting if you run it repeatedly (e.g. during development).

### CODE REVIEW

File: src/Commands/GitHub/SyncLabels.php [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+        $this->supportedModuleLoader = $supportedModuleLoader;
+        $this->github = $github;
+    }
+
+    protected function configureOptions()
+    {
+        $this->addArgument(
+            'modules',
+            InputArgument::OPTIONAL,
```
Comment (dhensby) [inline src/Commands/GitHub/SyncLabels.php]: you can have an array of arguments, I think? Which is a bit nicer than comma delimited

File: src/Application.php [hunk 2]
```diff
@@ -55,6 +57,10 @@ protected function getDefaultCommands()
     {
         $commands = parent::getDefaultCommands();
 
+        // Create dependencies
+        $githubApi = new GitHubApi();
```
Comment (dhensby) [inline src/Application.php]: If we are concerned about API limits we should employ a caching layer as cached API requests don't impact on the API rate limit

Comment (dhensby) [reply src/Application.php]: I'd also add a GitHubApi getter onto the commands that need it rather than instantiating it with the application

Comment (robbieaverill) [reply src/Application.php]: Yeah? I'd thought I'd go with dependency injection here. Happy to change though

Comment (dhensby) [reply src/Application.php]: Well, DI would be cool, but can we do it lazily?

Comment (robbieaverill) [reply src/Application.php]: This particular dependency doesn't do anything until you call `$githubApi->getClient()`, so I guess it's kind of lazy :D

File: src/Commands/GitHub/SyncLabels.php (hunk 1 above)
Comment (robbieaverill) [reply src/Commands/GitHub/SyncLabels.php]: TIL

**Comment by tractorcow** [comment]
Looks cool. :) Sorry I don't have time to do a full review.

**Pull request merged**